core/src/View/Kokkos_ViewTraits.hpp

```diff
@@ -488,20 +488,37 @@ struct ViewTraits {
  public:
   //------------------------------------
   // Data type traits:
-
+#ifdef KOKKOS_ENABLE_IMPL_VIEW_LEGACY
+  using data_type           = typename data_analysis::type;
+  using const_data_type     = typename data_analysis::const_type;
+  using non_const_data_type = typename data_analysis::non_const_type;
+#else
   using data_type           = typename data_analysis::data_type;
   using const_data_type     = typename data_analysis::const_data_type;
   using non_const_data_type = typename data_analysis::non_const_data_type;
+#endif
 
 #ifdef KOKKOS_ENABLE_DEPRECATED_CODE_5
   //------------------------------------
   // Compatible array of trivial type traits:
+#ifdef KOKKOS_ENABLE_IMPL_VIEW_LEGACY
```

Comment:
Probably move this outside of  the  `KOKKOS_ENABLE_DEPRECATED_CODE_5` block as currently `KOKKOS_ENABLE_DEPRECATED_CODE_5` and `KOKKOS_ENABLE_IMPL_VIEW_LEGACY` are orthogonal
.